Project21/Project21/Mealy.c

```diff
@@ -0,0 +1,116 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+
+// Define states of the Mealy machine
+typedef enum {
+   S0,  // Initial state
+   S1,  // After '0'
+   S2,  // After '01'
+   S3,  // After '011'
+   A1,
+   A2,
+   A3
+} State;
+
+void Mealy (char inpPath[], char outPath[]);
+State NextMealyState (State currentState, int input, int* output);
+
+void main (int argc, char* argv[]) {
+   char inpFilePath[MAX_PATH], outFilePath[MAX_PATH];
+   strcpy (inpFilePath, argv[1]);
+   strcpy (outFilePath, argv[2]);
+   Mealy (inpFilePath, outFilePath);
+}
+
+// Function to open input file and create a new file and store the result
+void Mealy (char inpPath[], char outPath[]) {
+   char ch, * tempPath = "temp_file.txt";
+   FILE* inpTestFile = fopen (inpPath, "r"),
+      * outTestFile = fopen (outPath, "r"),
+      * tempFile = fopen (tempPath, "w");
+   if (inpTestFile == NULL || outTestFile == NULL || tempFile == NULL) {

```

Comment:
inputTestFile == null can be simplified as !inputTestFile. Similarly for other two conditions